src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
+    if (requestSecret && requestSecret === secret) {
+      return true;
+    } else {
+      throw new UnauthorizedException('Invalid feedback secret');
+    }
+  }
+}
```

Comment:
PrimathonAICodeReviewer

> Note: This review was outside of the patch, but no patch was found that overlapped with it. Original lines [26-26]

The guard could benefit from logging the unauthorized access attempts for auditing purposes.
```diff
+     console.log('Unauthorized access attempt with invalid feedback secret');
```


<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->